### PR TITLE
feat: add per-delegate max_turns override

### LINKED ISSUE #6198: feat: Add max_turns to recipe and subagent settings
**Please explain the motivation behind the feature request.**

Subagents have a default max_turns limit of 25. Complex tasks (e.g., multi-file implementations with build/test verification) can exceed this limit, causing the subagent to stop with "I've reached the maximum number of actions I can do without user input."

Currently, the only way to configure this is via the `GOOSE_SUBAGENT_MAX_TURNS` environment variable, which:
- Requires external setup before running goose
- Applies globally to all subagents
- Cannot be shared as part of a recipe

**Describe the solution you'd like**

Add `max_turns` field to:

1. `Settings` struct in `crates/goose/src/recipe/mod.rs` - for top-level recipe configuration
2. `SubagentSettings` struct in `crates/goose/src/agents/subagent_tool.rs` - for per-subagent configuration

This would allow recipes to specify:
```yaml
settings:
  goose_provider: "gcp_vertex_ai"
  goose_model: "claude-sonnet-4-5@20250929"
  max_turns: 50
```

And subagent tool calls to override:
```json
{
  "settings": {
    "model": "claude-haiku-4-5@20251001",
    "max_turns": 30
  }
}
```

**Describe alternatives you've considered**

Setting `GOOSE_SUBAGENT_MAX_TURNS=50` in shell config. This works but is not portable across machines or shareable in recipes.

**Additional context**

Related: #3938 requested `max turns` for subagents but was closed before implementation. Since then, `SubagentSettings` was added with `provider`, `model`, `temperature` - adding `max_turns` would complete this.

Relevant code:
- `crates/goose/src/agents/subagent_task_config.rs` - defines `DEFAULT_SUBAGENT_MAX_TURNS` (25) and reads env var
- `crates/goose/src/agents/subagent_tool.rs` - `apply_settings_overrides` would need to handle `max_turns`

- [x] I have verified this does not duplicate an existing feature request

### PR DESCRIPTION
**Pull request by Yagi-Michael**
## Summary
- Adds a `max_turns` parameter to the delegate tool, allowing individual delegations to override the max turns limit
- Fixes priority order so delegated recipe `settings.max_turns` and session recipe are both respected
- Full priority chain: per-delegate `max_turns` > delegated recipe settings > session recipe > env var > config > default

## Context
The original `max_turns` implementation landed in `subagent_tool.rs`, but the summon refactor replaced that file with the unified summon extension (`platform_extensions/summon.rs`). The per-delegate override was never carried over, so it needs to be wired up in the new location.

Closes #6198

## Changes
- `DelegateParams`: new optional `max_turns: Option<usize>` field
- JSON schema: exposes `max_turns` as integer with `minimum: 1`
- Validation: rejects `max_turns: 0`
- `build_task_config`: checks `params.max_turns`, then delegated `recipe.settings.max_turns`, then falls back to `resolve_max_turns`
- `resolve_max_turns`: reordered to session recipe > env var (was env var > recipe)

## Test plan
- [x] Deserialization tests for `max_turns` presence and absence
- [x] Validation tests rejecting zero and accepting positive values
- [x] Resolution tests verifying recipe overrides env var
- [x] Resolution tests verifying env var fallback when no recipe
- [x] Resolution tests verifying default fallback
- [x] Env var tests use `#[serial]` for thread safety

### CODE REVIEW

File: crates/goose/src/agents/platform_extensions/summon.rs [hunk 1]
```diff
@@ -1536,7 +1541,10 @@ impl SummonClient {
             }
         }
 
-        let max_turns = self.resolve_max_turns(session);
+        // Priority: per-delegate max_turns > recipe settings > env var > config > default
+        let max_turns = params
+            .max_turns
+            .unwrap_or_else(|| self.resolve_max_turns(session));
```
[the remark this answers is not in the document]
Comment (Yagi-Michael) [reply crates/goose/src/agents/platform_extensions/summon.rs:1548]: Helpful feedback. Think this is real and not just fluff. Taking a look (using goose to help :goose: )

[the remark this answers is not in the document]
Comment (Yagi-Michael) [reply crates/goose/src/agents/platform_extensions/summon.rs:1548]: Good catch — this was a real bug. `resolve_max_turns` only checked `session.recipe` (the parent session's recipe), so a delegated source recipe's `settings.max_turns` was silently ignored.

Fixed by inserting an `.or_else()` check on the delegated `recipe.settings.max_turns` between the per-delegate `params.max_turns` override and the `resolve_max_turns(session)` fallback. The priority chain is now:

1. `params.max_turns` (per-delegate override)
2. Delegated recipe's `settings.max_turns`
3. Session recipe / env var / config / default

File: crates/goose/src/agents/platform_extensions/summon.rs [hunk 2]
```diff
@@ -1536,7 +1541,16 @@ impl SummonClient {
             }
         }
 
-        let max_turns = self.resolve_max_turns(session);
+        // Priority: per-delegate max_turns > delegated recipe settings > session recipe > env var > config > default
+        let max_turns = params
+            .max_turns
+            .or_else(|| {
+                recipe
+                    .settings
```
[the remark this answers is not in the document]
Comment (Yagi-Michael) [reply crates/goose/src/agents/platform_extensions/summon.rs]: Fixed — added `.clamp(1, u32::MAX as usize)` to the resolved max_turns before passing to `with_max_turns`. This handles both issues: rejects 0 from delegated recipe settings, and prevents wrapping when narrowed to u32 in `subagent_handler.rs:178`.

**Comment by Yagi-Michael** [comment]
:goose: please let me know how this looks or if would like any changes :goose:

File: crates/goose/src/agents/platform_extensions/summon.rs [hunk 3]
```diff
@@ -1536,7 +1541,12 @@ impl SummonClient {
             }
         }
 
-        let max_turns = self.resolve_max_turns(session);
+        // Priority: per-delegate max_turns > delegated recipe settings > session recipe > env var > config > default
+        let max_turns = params
+            .max_turns
+            .or_else(|| recipe.settings.as_ref().and_then(|s| s.max_turns))
+            .unwrap_or_else(|| self.resolve_max_turns(session))
+            .clamp(1, u32::MAX as usize);
```
[the remark this answers is not in the document]
Comment (Yagi-Michael) [reply crates/goose/src/agents/platform_extensions/summon.rs]: Okay codex, one more time :)

**Review by DOsinga — APPROVED** [review]
Good fix — addresses a real bug (delegated recipe `max_turns` was silently ignored) and a real edge case (u32 overflow). Nice job engaging with codex feedback and iterating. I cleaned up a few restatement comments and dropped two trivial deserialization tests (they only assert struct layout, not behaviour) in a follow-up commit.

**Comment by Yagi-Michael** [comment]
Woo, thanks! Thank you for the cleanup, will be better at catching those bloating going forward

**Comment by Yagi-Michael** [comment]
> Good fix — addresses a real bug (delegated recipe `max_turns` was silently ignored) and a real edge case (u32 overflow). Nice job engaging with codex feedback and iterating. I cleaned up a few restatement comments and dropped two trivial deserialization tests (they only assert struct layout, not behaviour) in a follow-up commit.

Hi @DOsinga, just curious what the merge cadence is and roughly when can this might get merged in
